tests/canfd/test_canfd.py

```diff
@@ -40,13 +39,42 @@ def panda_init(serial, enable_canfd=False, enable_non_iso=False):
   p = Panda(serial=serial)
   p.set_power_save(False)
   for bus in range(3):
+    p.set_can_speed_kbps(0, 500)
     if enable_canfd:
       p.set_can_data_speed_kbps(bus, 2000)
     if enable_non_iso:
       p.set_canfd_non_iso(bus, True)
   p.set_safety_mode(Panda.SAFETY_ALLOUTPUT)
   return p
 
+def test_canfd_throughput(p, p_recv=None):
+  two_pandas = p_recv is not None
+  p.set_safety_mode(Panda.SAFETY_ALLOUTPUT)
+  if two_pandas:
+    p_recv.set_safety_mode(Panda.SAFETY_ALLOUTPUT)
+  # enable output mode
+  else:
+    p.set_can_loopback(True)
+
+  tests = [
+    [500, 1000, 2000], # speeds
+    [93, 87, 78], # saturation thresholds
+  ]
+
+  for i in range(len(tests[0])):
+    # set bus 0 data speed to speed
+    p.set_can_data_speed_kbps(0, tests[0][i])
+    if p_recv is not None:
+      p_recv.set_can_data_speed_kbps(0, tests[0][i])
+    time.sleep(0.05)
+
+    comp_kbps = time_many_sends(p, 0, p_recv=p_recv, msg_count=400, two_pandas=two_pandas, msg_len=64)
+
+    # bit count from https://en.wikipedia.org/wiki/CAN_bus
+    saturation_pct = (comp_kbps / tests[0][i]) * 100.0
```

Comment:
should probably be more specific about the overall throughput, not just the data speed

Reply:
makes sense, will fix. needed mostly to have thresholds before I will lower TX FIFO size (similar to F4) for more control and less losses.